### ISSUE TITLE
Bluetooth Disconnects

### ISSUE THREAD

**Issue by InstallGentoo11** [Ryochan7/DS4Windows#276]
I am using DS4 v2 in Windows 7.

It seems like the bluetooth disconnects have been a thing for a while now, but I don't see an active thread for them and they have started becoming even more problematic as of updating to 118. Before it seemed to only disconnect me if I am only not active for a while (not the exact same time, though), but now I am getting disconnected even while playing sometimes. The logs say nothing, just that the controller is "unplugged". After getting to 118 sometimes I even have to stop and restart DS4Windows for it to be able find the controller when I am trying to connect it.
Since I changed from the original DS4Windows to this fork I started getting some red flashes, meaning 20+ latency, which I didn't get before either. It is probably related.

**Comment by jonaaa20** [comment]
Using a CSR Adapter, no problems here, I'm using the windows generic driver btw, so... Maybe is your Bluetooth card/adapter?

**Comment by aidenn** [comment]
Using Windows 10, a IOGear GBU421 and IOGear drivers I get 4 ms latency and no disconnects at all. Maybe try different drivers or Toshiba stack (https://wiki.dolphin-emu.org/index.php?title=Wii_Remote_Plus_(RVL-CNT-01-TR)_Connection_Guide#Toshiba_Stack).

That said, it's weird that you're getting different connection drop rates depending on the version of DS4Windows since AFAIK it doesn't negotiate the connection itself (it's handled by the OS). Do you get proper behaviour if you downgrade to previous DS4Windows now?

**Comment by Ryochan7** [comment]
Hopefully you have better luck with 1.4.119. Sometimes problems come from my changes but Windows can cause it own set of problems. I had changed how output reports were generated a bit with version 1.4.117 and 1.4.118. Some of those changes have been reverted lately.

BT connections work for me so I can't do much until I experience problems.

**Comment by InstallGentoo11** [comment]
I can almost definitely say it is not a DS4Windows issue.
I tried a different DS4 and got the same disconnect. I found there were newer drivers and installed them. Mind you I had done that before. I guess there was some driver issues, because the disconnects stopped happening. But I started getting some very often 10+ latency. I disabled the wifi capabilities of the lan card. And that is gone. I don't use them anyway, so it is not a big lose, but kind of sucks I have to disable something I have paid for. And the other lan card I have on another PC, which is suppose to even be cheaper is having both the wifi and bluetooh working no problem. It is even working as a hotspot that one.
But yeah, everything points to some driver issue or maybe hardware probably, but I really think Intel have just fucked up with the drivers.
